packages/caliper-core/test/rate-control/linearRate.js

```diff
@@ -0,0 +1,179 @@
+            const start = null;
+            const idx = 5;
+
+            controller._interpolateFromIndex(start, idx).should.equal(75);
+        });
+
+        it('should determine the interpolated value based on duration of the round', () => {
+            let msg = {
+                totalClients: 6,
+                txDuration: 5
+            };
+            controller.init(msg);
+            const start = 5;
+            const idx = null;
+
+            controller._interpolateFromTime(start, idx).should.equal(300.225);
+        });
+
+    });
+
+    describe('#applyRateController', () => {
+
+        let sleepStub;
+
+        beforeEach(() => {
+            sleepStub = sinon.stub();
+            LinearRate.__set__('util.sleep', sleepStub);
+
+            controller = new LinearRate.createRateController(opts);
+        });
+
+        it('should sleep for a duration of current sleep time, interpolated from index, if greater than 5ms', () => {
+            const currentSleepTime = 6;
+            let interpolateStub = sinon.stub().returns(currentSleepTime);
+            controller._interpolateFromIndex = interpolateStub;
+
+            let msg = {
+                totalClients: 6,
+                numb: 5
+            };
+            controller.init(msg);
+            const start = null;
+            const idx = 5;
+
+            controller.applyRateControl(start, idx, null, null);
+            sinon.assert.calledOnce(sleepStub);
+            sinon.assert.calledWith(sleepStub, currentSleepTime);
+        });
+        
+        it('should sleep for a duration of current sleep time, interpolated from time, if greater than 5ms', () => {
+            const currentSleepTime = 6;
+            let interpolateStub = sinon.stub().returns(currentSleepTime);
+            controller._interpolateFromTime = interpolateStub;
+
+            let msg = {
+                totalClients: 5,
+                txDuration: 5
+            };
+            controller.init(msg);
+            const start = 5;
+            const idx = null;
+
+            controller.applyRateControl(start, idx, null, null);
+            sinon.assert.calledOnce(sleepStub);
+            sinon.assert.calledWith(sleepStub, currentSleepTime);
+        });
+
+        it('should do nothing where current sleep time, interpolated from index, is less than or equal to 5ms', () => {
+            let currentSleepTime = 4;
+            let interpolateStub = sinon.stub().returns(currentSleepTime);
+            controller._interpolateFromIndex = interpolateStub;
+
+            let msg = {
+                totalClients: 5,
+                numb: 5
+            };
+            controller.init(msg);
+            const start = null;
+            const idx = 5;
+
+            controller.applyRateControl(start, idx, null, null);
+            sinon.assert.notCalled(sleepStub);
+        });
+
+        it('should do nothing where current sleep time, interpolated from time, is less than or equal to 5ms', () => {
```

Comment:
Since you mock `_interpolate`, this test is the same as the previous one (same for the >5ms sleep tests). That is when you use white-box testing. For black-box testing, the separate time/index-based tests are fine. 
Just an observation, not a change request, it's up to you which methodology you follow :)